src/utils/Validator.ts

```diff
@@ -182,26 +180,26 @@ const validateTimestamp = (timestamp: Date | number): void => {
     throw new InvalidTimestampError(`Timestamp ${String(timestamp)} is invalid`);
   }
 
-  const isTooOld = validator.isBefore(
-    timestampStr,
-    new Date(Date.now() - Constants.MAX_TIMESTAMP_PAST_AGE).toISOString(),
-  );
-  const isTooNew = validator.isAfter(
-    timestampStr,
-    new Date(Date.now() + Constants.MAX_TIMESTAMP_FUTURE_AGE).toISOString(),
-  );
+  if (!isDate(timestampStr)) {
```

Comment:
```
/**
 * Validates timestamp.
 * @see [CloudWatch Timestamp](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/cloudwatch_concepts.html#about_timestamp)
 *
 * @param timestamp
 */
const validateTimestamp = (timestamp: Date | number): void => {
  if (!isDate(timestamp)) {
    throw new InvalidTimestampError(`Timestamp ${String(timestamp)} is invalid`);
  }

  if (timestamp < new Date(Date.now() - Constants.MAX_TIMESTAMP_PAST_AGE)) {
    throw new InvalidTimestampError(
      `Timestamp ${timestamp} must not be older than ${Constants.MAX_TIMESTAMP_PAST_AGE} milliseconds`,
    );
  }

  if (timestamp > new Date(Date.now() + Constants.MAX_TIMESTAMP_FUTURE_AGE)) {
    throw new InvalidTimestampError(
      `Timestamp ${timestamp} must not be newer than ${Constants.MAX_TIMESTAMP_FUTURE_AGE} milliseconds`,
    );
  }
};

const isDate = (timestamp: Date | number): boolean => {
  return timestamp instanceof Date && new Date(timestamp).getTime() > 0;
};
```

Reply:
I was trying to do something similar to `ToDate` of validator package. But yes, this makes more sense and it's easy to understand. Thanks a lot! Updated the code.

Reply:
I made a mistake in this block of code - Fixed in a new commit to this PR - Also added tests that cover success cases for timestamp